core/src/main/java/org/mobilitydata/gtfsvalidator/input/GtfsInput.java

```diff
@@ -56,11 +63,100 @@ public static GtfsInput createFromPath(Path path) throws IOException {
+            new InvalidInputFilesInSubfolderNotice(invalidInputMessage));
+      }
+      return new GtfsZipFileInput(zipFile);
     }
     // Load a remote ZIP file to memory.
-    return new GtfsZipFileInput(
-        new ZipFile(new SeekableInMemoryByteChannel(Files.readAllBytes(path))));
+    zipFile = new ZipFile(new SeekableInMemoryByteChannel(Files.readAllBytes(path)));
+    if (containsSubfolderWithGtfsFile(path)) {
+      noticeContainer.addValidationNotice(
+          new InvalidInputFilesInSubfolderNotice(invalidInputMessage));
+    }
+    return new GtfsZipFileInput(zipFile);
+  }
+
+  /**
+   * Check if a zip file contains a subfolder with GTFS files
+   *
+   * @param path
+   * @return
+   */
+  public static boolean containsSubfolderWithGtfsFile(Path path) {
+    boolean containsGtfsFileInSubfolder = false;
+    boolean containsSubfolder = false;
+    String subfolder = null;
+    try (ZipInputStream zipInputStream =
+        new ZipInputStream(new BufferedInputStream(new FileInputStream(path.toFile())))) {
+      ZipEntry entry;
+      while ((entry = zipInputStream.getNextEntry()) != null) {
+        String entryName = entry.getName();
+        if (entry.isDirectory()) {
+          containsSubfolder = true;
+          subfolder = entry.getName();
+        }
+        if (containsSubfolder && entryName.contains(subfolder) && entryName.endsWith(".txt")) {
+          String[] files = entryName.split("/");
+          String lastElement = files[files.length - 1];
+          if (containsGtfsFile(lastElement)) {
+            containsGtfsFileInSubfolder = true;
+          }
+        }
+      }
+    } catch (IOException ioException) {
+      logger.atSevere().withCause(ioException).log(
+          "GtfsInput containsSubfolderWithTxtFile throws IOException");
+    } finally {
+      return containsGtfsFileInSubfolder;
+    }
+  }
+
+  public static boolean containsSubfolderWithGtfsFile(URL url) {
+    boolean containsGtfsFileInSubfolder = false;
+    boolean containsSubfolder = false;
+    String subfolder = null;
+    try (ZipInputStream zipInputStream =
+        new ZipInputStream(new BufferedInputStream(url.openStream()))) {
+      ZipEntry entry;
+      while ((entry = zipInputStream.getNextEntry()) != null) {
+        String entryName = entry.getName();
+        if (entry.isDirectory()) {
+          containsSubfolder = true;
+          subfolder = entry.getName();
+        }
+        if (containsSubfolder && entryName.contains(subfolder) && entryName.endsWith(".txt")) {
+          String[] files = entryName.split("/");
+          String lastElement = files[files.length - 1];
+          if (containsGtfsFile(lastElement)) {
+            containsGtfsFileInSubfolder = true;
+          }
+        }
+      }
+    } catch (IOException ioException) {
+      logger.atSevere().withCause(ioException).log(
+          "GtfsInput containsSubfolderWithTxtFile throws IOException");
+    } finally {
+      return containsGtfsFileInSubfolder;
```

Comment:
I think it's ok. If an IOException is thrown, that means we cannot read the gtfs input files properly, be it file not found or file path is incorrect. In such cases, it's ok to return false coz we haven't even read the input file.

Reply:
Ok I see, then I can remove the finally block and put the return statement at the of the method.

Reply:
One last comment about this: To be perfectly safe you should catch Exception instead of IOException. That way you will be sure the method does not send any exception.

Reply:
Actually that's a very good trick about writing try/catch block! I'll keep it in mind. And as it's optional, I'll keep the code as is. I'll go ahead merge it into master.